src/engine/strat_engine/backstore/metadata/mda.rs

```diff
@@ -367,12 +362,38 @@ impl MDARegions {
     }
 }
 
+#[derive(Debug, Default, PartialEq, PartialOrd)]
+pub struct MetaDataSize(Bytes);
+
+impl MetaDataSize {
+    pub fn new(used: Bytes, max_available: Bytes) -> StratisResult<MetaDataSize> {
```

Comment:
I think it's best to not have the ```max_available``` parameter here. It made sense to have a minimum for MDADataSize, but the minimum for that is a constant, so doesn't need to be a parameter. Here, though, the minimum for the MetaDataSize depends on the value for the MDADataSize, which is not a constant. Better to just construct the value as it is, and leave the comparison check in ```save_state```.